### ISSUE TITLE
nans obtained when training the model with very different pre and post dataset sizes

### ISSUE THREAD

**Issue by adigioacchino** [statbiophys/soNNia#1]
## Description of the issue:
When training the model with n_pre "pre data" and n_post "post data", if n_pre >> n_post or n_post >> n_pre, the training completes without any error, but the loss and the likelihood values (of both the training and validation sets) are all nans. 

## Minimal example of the issue:
In this example I generate the pre and post sequences using soNNia itself, but the same happened with my actual dataset. I only show the case with n_pre >> n_post, but also the case  n_pre << n_post gives the nans.
```
from sonnia.sonnia import SoNNia

# instantiate model
qm = SoNNia()

# prepare some example sequences
n_tot = 100000
qm.add_generated_seqs(num_gen_seqs = n_tot, reset_gen_seqs = True)

# prepare pre and post data
n_pre = 100000
n_post = 1000
seqs_post = qm.gen_seqs[:n_post]
seqs_pre = qm.gen_seqs[-n_pre:]

# model training (here I get nans)
qm.infer_selection(epochs = 20, batch_size = 1000, validation_split=0.05, verbose=1)
```

## Error message:
No specific error message is given when the nans appear in the printed output of qm.infer_selection (with verbose=1).
